src/strategies/offer_maker/abstract/Direct.sol

```diff
@@ -77,27 +65,22 @@ abstract contract Direct is MangroveOffer {
       return; // nothing to do
     } else {
       // noop if reserve == address(this)
-      router_.push(token, reserve(), amount);
+      router_.push(token, reserve(admin()), amount);
     }
   }
 
-  function tokenBalance(IERC20 token) external view override returns (uint) {
-    AbstractRouter router_ = router();
-    return router_ == NO_ROUTER ? token.balanceOf(reserve()) : router_.reserveBalance(token, reserve());
-  }
-
   function flush(IERC20[] memory tokens) internal {
     AbstractRouter router_ = MOS.getStorage().router;
     if (router_ == NO_ROUTER) {
       for (uint i = 0; i < tokens.length; i++) {
         require(
```

Comment:
Why not call `reserve(admin())` outside for loop. Doing it inside for loop, could result in multiple calls